src/domain/youthProfile/edit/EditYouthProfile.tsx

```diff
@@ -43,28 +38,14 @@ function EditYouthProfile(props: Props) {
     }
   );
 
-  const [updateProfile, { loading: saveLoading }] = useMutation<
-    UpdateMyProfileData,
-    UpdateMyProfileVariables
-  >(UPDATE_PROFILE);
+  const [updateProfiles, { loading }] = useUpdateProfiles({
+    onError: () => setShowNotification(true),
+  });
 
-  const history = useHistory();
   const youthProfile = data?.myYouthProfile;
 
   const handleOnValues = (formValues: FormValues) => {
-    const variables: UpdateMyProfileVariables = getEditMutationVariables(
-      formValues,
-      data
-    );
-
-    updateProfile({ variables })
-      .then(() => {
-        history.push('/membership-details');
-      })
-      .catch((error: Error) => {
-        Sentry.captureException(error);
-        setShowNotification(true);
-      });
+    updateProfiles(formValues, data);
```

Comment:
I think having the redirect in this level is better separation of concerns. Some consumers of the hook may want to redirect somewhere else than this view does. Generally, doing the redirect here allows the hook to be agnostic about the url structure of this application for instance.